reconcile/utils/slack_api.py

```diff
@@ -16,36 +20,84 @@ class UsergroupNotFoundException(Exception):
     pass
 
 
-class SlackAPICallException(Exception):
-    """Raised for general error cases when calling the Slack API."""
-
-
-class SlackAPIRateLimitedException(SlackAPICallException):
-    """Raised when a call to the Slack API has been rate-limited."""
+class ServerErrorRetryHandler(RetryHandler):
+    """Retry handler for 5xx errors."""
+    def _can_retry(
+        self,
+        *,
+        state: RetryState,
+        request: HttpRequest,
+        response: Optional[HttpResponse] = None,
+        error: Optional[Exception] = None
+    ) -> bool:
+        return response is not None and response.status_code >= 500
 
 
 class SlackApi:
     """Wrapper around Slack API calls"""
 
-    def __init__(self, workspace_name, token,
-                 settings=None,
+    def __init__(self,
+                 workspace_name: str,
+                 token: Mapping[str, str],
+                 settings: Optional[Mapping[str, Any]] = None,
                  init_usergroups=True,
-                 **chat_kwargs):
+                 channel: Optional[str] = None,
```

Comment:
Why is this one optional?

Reply:
I'll update docstring and we can address again if still unclear.